inferno/tests/test_dataset.py

```diff
@@ -364,3 +389,276 @@ def test_fit_predict_proba(self, net, data):
         net.fit(X, y)
         y_proba = net.predict_proba(X)
         assert np.allclose(y_proba.sum(1), 1)
+
+
+class TestTrainSplitIsUsed:
+    @pytest.fixture
+    def iterator(self):
+        class Iterator:
+            """An iterator that just yield the input data."""
+            def __init__(self, dataset, *args, **kwargs):
+                self.dataset = dataset
+
+            def __iter__(self):
+                yield self.dataset.X, self.dataset.y
+
+        return Iterator
+
+    @pytest.fixture
+    def data(self):
+        X = torch.arange(0, 12).view(4, 3)
+        y = torch.LongTensor([0, 1, 1, 0])
+        return X, y
+
+    @pytest.fixture
+    def data_split(self, data):
+        X, y = data
+        return X[:2], X[2:], y[:2], y[2:]
+
+    @pytest.fixture
+    def module(self):
+        class MyClassifier(nn.Module):
+            def __init__(self, input_units=20, num_units=10, nonlin=F.relu):
+                super(MyClassifier, self).__init__()
+
+                self.dense0 = nn.Linear(input_units, num_units)
+                self.nonlin = nonlin
+                self.dropout = nn.Dropout(0.5)
+                self.dense1 = nn.Linear(num_units, 10)
+                self.output = nn.Linear(10, 2)
+
+            def forward(self, X, **kwargs):
+                X = self.nonlin(self.dense0(X))
+                X = self.dropout(X)
+                X = F.relu(self.dense1(X))
```

Comment:
Why not use the configured nonlin here as well? OTOH why have it configurable at all?